src/sentry/integrations/gitlab/repository.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import
+
+import six
+
+from sentry.plugins import providers
+from sentry.models import Integration
+
+
+class GitlabRepositoryProvider(providers.IntegrationRepositoryProvider):
+    name = 'Gitlab'
+
+    def get_installation(self, integration_id, organization_id):
+        if integration_id is None:
+            raise ValueError('%s requires an integration_id' % self.name)
+
+        try:
+            integration_model = Integration.objects.get(id=integration_id)
```

Comment:
We can, but I'd need to do this differently.`OrganizationIntegration` holds the org not the  `Integrations` model. It's a many to one relationship. This function is used in an api endpoint. I think it's probably much safer to do it the way you're suggesting. With the caveat that it was done this way in multiple integrations (because .... I was in charge of more than one and never really got this feedback :( ). We should change it; But err... maybe some refactoring is in order in the other integrations I built?

Reply:
**Please add this**

If we're not filtering to organization on API endpoints we definitely need to be. You can use djangos ORM magic here and say:

```py
integration_model = Integration.objects.get(
  id=integration_id,
  organizations=organization_id
)
```

that will magically join the `orgnization_integrations` and make sure it exists.

Reply:
@EvanPurkhiser I know this is silly to comment on, but I'm kind of opposed to this magic. It just doesn't make a ton of sense to me conceptually. I'd be ok with `organizations__id__in=[organization_id]` working, but feels like that's a step too far. Again just might be me feeling weird about how that works. :) Thank you for mentioning it.